### PR TITLE
Mshare 100 add date to spending

### PR DESCRIPTION
**Pull request by horvath-martin**

### CODE REVIEW

**Comment by hrvthp** [comment]
Can one of the admins verify this patch?

**Review by g-jozsef — APPROVED** [review]
Backend OK

File: frontend/android/app/src/main/java/elte/moneyshare/util/TimeExtensions.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package elte.moneyshare.util
+
+import java.text.DateFormat
+import java.text.SimpleDateFormat
+import java.util.TimeZone
+import java.util.Calendar
+
+class TimeExtensions {
```
Comment (atus89123) [inline frontend/android/app/src/main/java/elte/moneyshare/util/TimeExtensions.kt]: rename to DateExtensions, and don't need to create class, simply you can create extensions methods to any type.
like this:
fun formatDate(calendar: Calendar): String = DateFormat.getDateInstance().format(calendar.time)
->
fun Calendar.formatDate(): String = DateFormat.getDateInstance().format(time)
"this" (this.time) will be the object on called this method, dont need plus paramter and the static class to use it

File: frontend/android/app/src/main/java/elte/moneyshare/view/AddSpendingFragment.kt [hunk 2]
```diff
@@ -85,6 +101,17 @@ class AddSpendingFragment : Fragment() {
             }
         }
 
+        dateEditText!!.setOnClickListener {
```
Comment (atus89123) [inline frontend/android/app/src/main/java/elte/moneyshare/view/AddSpendingFragment.kt]: replace !! to ? to do nullcheck in safe way

File: frontend/android/app/src/main/res/layout/fragment_add_spending.xml [hunk 3]
```diff
@@ -34,12 +34,27 @@
             android:layout_marginLeft="12dp"
             android:layout_marginRight="12dp"/>
 
+    <android.support.v7.widget.AppCompatEditText
+            android:id="@+id/dateEditText"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="xxxx.xx.xx."
```
Comment (atus89123) [inline frontend/android/app/src/main/res/layout/fragment_add_spending.xml]: use hint, like other edittexts

File: frontend/android/app/src/main/res/layout/fragment_add_spending.xml [hunk 4]
```diff
@@ -34,12 +34,27 @@
...
+            app:layout_constraintTop_toBottomOf="@+id/spendingEditText"
+            app:layout_constraintLeft_toLeftOf="parent"
+            app:layout_constraintRight_toRightOf="parent"
+            android:layout_marginTop="12dp"
+            android:layout_marginLeft="12dp"
+            android:layout_marginRight="12dp"
+            android:clickable="true"
+            android:focusable="false"
+            android:inputType="textNoSuggestions"/>
```
Comment (atus89123) [inline frontend/android/app/src/main/res/layout/fragment_add_spending.xml:50]: try date input type maybe, can we use something to add better userexperince to users to know its a date, add label (TextView) or something next to edittext?

**Pull request merged**